depthai/plugin.cpp

```diff
@@ -0,0 +1,344 @@
+        , _m_imu_cam{sb->get_writer<imu_cam_type>("imu_cam")}
+        , _m_rgb_depth{sb->get_writer<rgb_depth_type>("rgb_depth")}
+        //Initialize DepthAI pipeline and device 
+        , p{createCameraPipeline()}
+        , d{p}
+        { 
+            d.startPipeline();
+            #ifndef NDEBUG
+                std::cout << "Depthai pipeline started" << std::endl;
+            #endif
+            color = d.getOutputQueue("preview", 1, false);
+            depthQueue = d.getOutputQueue("depth", 1, false);
+            rectifLeftQueue = d.getOutputQueue("rectified_left", 1, false);
+            rectifRightQueue = d.getOutputQueue("rectified_right", 1, false);
+            imuQueue = d.getOutputQueue("imu", 1, false);
+            std::function<void(void)> imu_callback = [&](){callback();};
+            imuQueue->addCallback(imu_callback);
+        }
+
+    void callback(){  
+        std::lock_guard<std::mutex> lock(mutex);
+        //Check for available data
+        bool color_go = color->has<dai::ImgFrame>();
+        bool depth_go = depthQueue->has<dai::ImgFrame>();
+        bool rectifL_go = rectifLeftQueue->has<dai::ImgFrame>();
+        bool rectifR_go = rectifRightQueue->has<dai::ImgFrame>();
+        bool imuPacket_go = imuQueue->has<dai::IMUData>();
+        
+        #ifndef NDEBUG
+                if(color_go){
+                    rgb_count++;
+                }
+                if(depth_go){
+                    depth_count++;
+                }
+                if(rectifL_go){
+                    left_count++;
+                }
+                if(rectifR_go){
+                    right_count++;
+                }
+        #endif
+
+        if(rectifR_go && rectifL_go && depth_go && color_go) {
+            #ifndef NDEBUG
+                all_count++;
+            #endif
+            auto colorFrame = color->tryGet<dai::ImgFrame>();
+            auto depthFrame = depthQueue->tryGet<dai::ImgFrame>();
+            auto rectifL = rectifLeftQueue->tryGet<dai::ImgFrame>();
+            auto rectifR = rectifRightQueue->tryGet<dai::ImgFrame>();
+            
+            cv::Mat color = cv::Mat(colorFrame->getHeight(), colorFrame->getWidth(), CV_8UC3, colorFrame->getData().data());
+            cv::Mat rgb_out{color.clone()};
+            cv::Mat rectifiedLeftFrame = cv::Mat(rectifL->getHeight(), rectifL->getWidth(), CV_8UC1, rectifL->getData().data());
+            cv::Mat LeftOut{rectifiedLeftFrame.clone()};
+            cv::flip(LeftOut, LeftOut, 1);
+            cv::Mat rectifiedRightFrame = cv::Mat(rectifR->getHeight(), rectifR->getWidth(), CV_8UC1, rectifR->getData().data());
+            cv::Mat RightOut{rectifiedRightFrame.clone()};
+            cv::flip(RightOut, RightOut, 1);
+
+            cv::Mat depth = cv::Mat(depthFrame->getHeight(), depthFrame->getWidth(), CV_16UC1, depthFrame->getData().data());
+            cv::Mat converted_depth;
+            depth.convertTo(converted_depth, CV_32FC1, 1000.f);
```

Comment:
I think `.convertTo` makes a clone, so need to `.clone` depth. This is important because cloning is probably expensive.

Reply:
Yes, it appears this use of `.convertTo` would trigger reallocation so no need for _another_ clone. From the docs

<!--StartFragment-->
m | output matrix; if it does not have a proper size or type before the operation, it is reallocated.
-- | --


<!--EndFragment-->

[Doc Ref](https://docs.opencv.org/master/d3/d63/classcv_1_1Mat.html#adf88c60c5b4980e05bb556080916978b)